src/ai/defaultai.cc

```diff
@@ -2926,6 +2971,18 @@ bool DefaultAI::construct_building(const Time& gametime) {
 					number_of_same_nearby += same_it->second;
 				}
 
+				// Considering distance to wh for various types of productionsites:
+				if (bo.is(BuildingAttribute::kWell) || bo.is(BuildingAttribute::kRanger) ||
```

Comment:
to satisfy codecheckk add || bo.is(BuildingAttribute::kSpaceConsumer) here to satisfy codecheck